components/eventing-controller/controllers/subscription/beb/reconciler_internal_test.go

```diff
@@ -14,17 +27,119 @@ import (
 
 	. "github.com/onsi/gomega"
 
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
 	reconcilertesting "github.com/kyma-project/kyma/components/eventing-controller/testing"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
 	domain = "domain.com"
 )
 
+// TestReconciler_Reconcile tests the return values of the Reconcile() method of the reconciler.
+// This is important, as it dictates whether the reconciliation should be requeued by Controller Runtime,
+// and if so with how much initial delay. Returning error or a `Result{Requeue: true}` would cause the reconciliation to be requeued.
+// Everything else is mocked since we are only interested in the logic of the Reconcile method and not the reconciler dependencies.
+func TestReconciler_Reconcile(t *testing.T) {
+	ctx := context.Background()
+	req := require.New(t)
+
+	// A subscription with the correct Finalizer, conditions and status ready for reconciliation with the backend.
+	testSub := reconcilertesting.NewSubscription("sub1", "test",
+		reconcilertesting.WithConditions(eventingv1alpha1.MakeSubscriptionConditions()),
+		reconcilertesting.WithFinalizers([]string{Finalizer}),
+		reconcilertesting.WithFilter(reconcilertesting.EventSource, reconcilertesting.OrderCreatedEventType),
+		reconcilertesting.WithValidSink("test", "test-svc"),
+		reconcilertesting.WithEmsSubscriptionStatusActive(),
+	)
+	// A subscription marked for deletion.
+	testSubUnderDeletion := reconcilertesting.NewSubscription("sub2", "test",
+		reconcilertesting.WithNonZeroDeletionTimestamp(),
+		reconcilertesting.WithFinalizers([]string{Finalizer}),
+		reconcilertesting.WithFilter(reconcilertesting.EventSource, reconcilertesting.OrderCreatedEventType),
+	)
+
+	backendSyncErr := errors.New("backend sync error")
+	backendDeleteErr := errors.New("backend delete error")
+	validatorErr := errors.New("invalid sink")
+	happyValidator := sink.ValidatorFunc(func(s *eventingv1alpha1.Subscription) error { return nil })
+	unhappyValidator := sink.ValidatorFunc(func(s *eventingv1alpha1.Subscription) error { return validatorErr })
+
+	var testCases = []struct {
+		name                 string
+		givenSubscription    *eventingv1alpha1.Subscription
+		givenReconcilerSetup func() *Reconciler
+		wantReconcileResult  ctrl.Result
+		wantReconcileError   error
+	}{
+		{
+			name:              "Return nil and default Result{} when there is no error from the reconciler dependencies",
+			givenSubscription: testSub,
```

Comment:
We use the same instance of `testSub` in multiple test cases, and IMO it is not future proof, because reconciling the same instance from multiple test-cases could lead to unwanted/unforeseen changes.

I suggest to either to initialise a new instance every time a test case is executed, or use a deepCopy from the original instance.

Reply:
I verified this and I can confirm that since we add the subscription `testSub` to the fake clientBuilder each time during a testcase, the subscription is as good as new.